### ISSUE TITLE
Navigate Up

### ISSUE THREAD

**Issue by codeguru42** [BaseballCardTracker/bbct#333]
Fix "up" navigation in the Action Bar.

**Resources:**
[Tasks and Back Stack][1]
[Intent.FLAG_ACTIVITY_CLEAR_TOP][2]

[1]:http://developer.android.com/guide/components/tasks-and-back-stack.html
[2]:http://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_CLEAR_TOP

**Comment by codeguru42** [comment]
Another Stack Overflow question:

[How to implement onBackPressed() in Fragments?][1]

[1]:

**Comment by jojonarte** [comment]
Can I take this?

**Comment by codeguru42** [comment]
@jojonarte Definitely. Be sure to include tests. Navigation might work on some screens, so it will probably help to write some tests first. This is just a suggestion, though. As long as the final pull request has tests, you do not have to follow strict Test Driven Development.

**Comment by codeguru42** [comment]
@jojonarte After you accept the invite to join the BBCT team on GitHub, I will officially add assign this issue to you. Or you can assign it to yourself.

**Comment by codeguru42** [comment]
To clarify this issue, clicking "up" from About, BaseballCardDetails, or FilterCards, should all return to BaseballCardList. In a previous version, FilterCards had a couple of subscreens. Without those screens, the "up" button should be equivalent to clicking Back.

**Comment by jojonarte** [comment]
Okay got it.

**Comment by jojonarte** [comment]
I think I'm done with implementing the fix for the up button, I'm currently writing the tests.

**Comment by codeguru42** [comment]
@jojonarte Good to hear! Create a pull request whenever you are ready. Be sure to create a new branch for your commits.

**Comment by jojonarte** [comment]
@codeguru42  Is BaseballCardDetails supposed to always be the first screen?

**Comment by codeguru42** [comment]
@jojonarte If there are no cards in the database, then the app will start on BaseballCardDetails. If there are, then the app will start on BaseballCardlist.

**Comment by jojonarte** [comment]
almost done with changes now , tests remaining

**Comment by codeguru42** [comment]
@jojonarte Yes, feel free to reuse `DataTestRule`. I encourage you to reuse any code you can. You can also refactor any code so that it is more reusable even if it is originally written for a single purpose.

**Comment by codeguru42** [comment]
@jojonarte Note that `DataTestRule` reads a file and accesses the database. This causes a huge impact to the amount of time required by any test suite that uses it. So be sure that it is absolutely required for the tests in that suite.
